Palaso/WritingSystems/LdmlDataMapper.cs

```diff
@@ -493,11 +493,33 @@ public void Write(string filePath, WritingSystemDefinition ws, Stream oldFile, W
 					readerSettings.ProhibitDtd = false;
 					reader = XmlReader.Create(oldFile, readerSettings);
 				}
-				using (var writer = XmlWriter.Create(filePath, CanonicalXmlSettings.CreateXmlWriterSettings()))
+				
+				string backupFilePath = null;
+				if (File.Exists(filePath))
 				{
-					writer.WriteStartDocument();
-					WriteLdml(writer, reader, ws);
-					writer.Close();
+					try
+					{
+						backupFilePath = Path.ChangeExtension(Path.GetTempFileName(), Path.GetExtension(filePath));
```

Comment:
There's a method on TempFile to get one with a particular extension. Using this would also make sure that the backup gets deleted.